src/python/dxpy/bindings/dxfile_functions.py

```diff
@@ -466,3 +466,50 @@ def upload_string(to_upload, media_type=None, keep_open=False, wait_on_close=Fal
         handler.close(block=wait_on_close, **remaining_kwargs)
 
     return handler
+
+def download_folder(project, destdir, folder="/", chunksize=dxfile.DEFAULT_BUFFER_SIZE):
```

Comment:
> Please allow the caller to supply additional kwargs, and pass them through to download_dxfile (download_dxfile itself does this). These can be used to customize the low-level request parameters.

Done

> Can this method be unified with existing code in dxpy for downloading folders? For example, "dx download -r" must do something similar, so possibly it could be expressed in terms of this method...

I tried to implement that but not successfully. I added an **overwrite** parameter to **download_folder()** method to provide the same behavior as **dx download -r** command. But after that I found, that **dx download -r** used a [directory tree cache](https://github.com/dnanexus/dx-toolkit/blob/master/src/python/dxpy/cli/download.py#L66) to optimize it's work if to download several remote folders. And it seems to be not a good idea to introduce a similar user-provided directory tree cache to **download_folder()** public API function. What do you think?

Thank you!